### ISSUE TITLE
False positive on map_clone

### ISSUE THREAD

**Issue by In-line** [rust-lang/rust-clippy#3551]
```rust
extern crate libc; 
use std::ffi::CStr;
use libc::c_char;

unsafe fn test(str_ptr: *const c_char) {
    let var: Vec<u8> =CStr::from_ptr(str_ptr)
        .to_bytes()
        .iter()
        .map(|i| *i)
        //.clone()
        .collect();

}

fn main() {
    unsafe {
       test(std::ptr::null());
    }
}
```
```rust
   Checking playground v0.0.1 (/playground)
warning: unused variable: `var`
 --> src/main.rs:6:9
  |
6 |     let var: Vec<u8> =CStr::from_ptr(str_ptr)
  |         ^^^ help: consider using `_var` instead
  |
  = note: #[warn(unused_variables)] on by default

warning: You are using an explicit closure for cloning elements
 --> src/main.rs:6:23
  |
6 |       let var: Vec<u8> =CStr::from_ptr(str_ptr)
  |  _______________________^
7 | |         .to_bytes()
8 | |         .iter()
9 | |         .map(|i| *i)
  | |____________________^
  |
  = note: #[warn(clippy::map_clone)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_clone
help: Consider calling the dedicated `cloned` method
  |
6 |     let var: Vec<u8> =CStr::from_ptr(str_ptr)
7 |         .to_bytes()
8 |         .iter().cloned()
  |

    Finished dev [unoptimized + debuginfo] target(s) in 0.84s
```
([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=41040656cd7f3235eff8cdb8b6b419a4))

**Comment by mikerite** [comment]
This seems to work for me. What error do you get? It looks like you might be trying to use `clone()` instead of `cloned()` as suggested.

**Comment by In-line** [comment]
Yeah, my bad autocomplete didn't show me `cloned`.
(with `clone()`)
```
error[E0277]: a collection of type `std::vec::Vec<u8>` cannot be built from an iterator over elements of type `&u8`
  --> src/lib.rs:11:10
   |
11 |         .collect();
   |          ^^^^^^^ a collection of type `std::vec::Vec<u8>` cannot be built from `std::iter::Iterator<Item=&u8>`
   |
   = help: the trait `std::iter::FromIterator<&u8>` is not implemented for `std::vec::Vec<u8>`
```

But error is non human oriented anyway..
